lib/links.rb

```diff
@@ -6,15 +6,16 @@ def initialize(url)
   end
 
   def run
-    ary = Parser.new(@url).run
+    links = Parser.new(@url).run
```

Comment:
How do you feel about creating the parser in the initializer? Seems to be a good flow.

```ruby
def initialize(url)
  @url = url
  @parser = Parser.new(url)
end

def run
  @parser.run.each do |link|
    # ...
   end
end
```

Reply:
Yes, makes no sense to create an instance of Links without a Parser